app/controllers/application_controller.rb

```diff
@@ -63,13 +63,18 @@ def self.action_auth_level(action, level)
 
   def self.action_no_auth(action)
     skip_before_action :verify_authenticity_token, :authenticate_user!
-    skip_filter configure_permitted_paramters: [action]
     skip_filter maintenance_mode: [action]
     skip_filter run_scheduler: [action]
 
     skip_filter authenticate_user: [action]
     skip_before_action :authorize_user_for_course, only: [action]
```

Comment:
I think we should break this up into 2 before_actions: set_course loads @course, and set_cud loads @cud/makes sure that the user is authorized to view the course.